ssl/record/rec_layer_s3.c

```diff
@@ -759,24 +759,31 @@ int do_ssl3_write(SSL *s, int type, const unsigned char *buf,
     /* Clear our SSL3_RECORD structures */
     memset(wr, 0, sizeof wr);
     for (j = 0; j < numpipes; j++) {
-        /* write the header */
-        *(outbuf[j]++) = type & 0xff;
-        SSL3_RECORD_set_type(&wr[j], type);
+        if (BIO_get_offload_tx(s->wbio)) {
+            /* IV will be generated by the kernel if offload is used */
+            eivlen = 0;
```

Comment:
It seems more appropriate for this to be set in the section of code above where eivlen is originally set.